kinto/core/views/swagger.py

```diff
@@ -64,15 +64,20 @@ def swagger_view(request):
 
     schemes = [settings.get('http_scheme') or 'http']
 
-    # Default security should match all security defs
-    auth_types = swagger_view.__json__.get('securityDefinitions', {})
-    security = swagger_view.__json__.get('security', [])
+    security_defs = swagger_view.__json__.get('securityDefinitions', {})
+
+    # BasicAuth is a non extension capability, so we should check it from config
+    if 'basicauth' in aslist(settings.get('multiauth.policies', '')):
+        basicauth = {'type': 'basic',
+                     'description': 'HTTP Basic Authentication.'}
+        security_defs['basicAuth'] = basicauth
 
-    # Security definitions are JSON objects with a single key
+    # Security options are JSON objects with a single key
+    security = swagger_view.__json__.get('security', [])
     security_names = [next(iter(security_def)) for security_def in security]
 
-    # Include securityDefinitions that are not on default security
-    for name, prop in auth_types.items():
+    # include securityDefinitions that are not on default security options
+    for name, prop in security_defs.items():
         security_def = {name: prop.get('scopes', {}).keys()}
```

Comment:
I think some more comments would be helpful here to readers who don't already know much about the OpenAPI spec. How about: `The 'security' field lists all acceptable ways of authenticating, so no plugin should add something to 'securityDefinitions' without it being here too`, and then `A 'security' requirement is an object of the form {security_mechanism: ['list', 'of', 'oauth2', 'scopes']}  # or an empty list if oauth2 isn't used`.